### ISSUE TITLE
False positive Magento2.Legacy.PhtmlTemplate

### ISSUE THREAD

**Issue by DmitryFurs** [magento/magento-coding-standard#351]
### Preconditions

1. `.phtml` file with `<script type="text/x-magento-init">`
2. magento-coding-standard v16

### Steps to reproduce

1. Create `.phtml` file with `<script type="text/x-magento-init">`
2. run `phpcs`

### Expected result

1. No `Magento2.Legacy.PhtmlTemplate` triggered

### Actual result

```FILE: /view/frontend/templates/product/variation.phtml
-------------------------------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
-------------------------------------------------------------------------------------------------------------------------------
 49 | WARNING | Please do not initialize JS component in php. Do it in template
-------------------------------------------------------------------------------------------------------------------------------
```

### Additional Notes
It's probably false positive triggering, or I'm missing something and now it's forbidden to use the `<script type="text/x-magento-init"></script>` script and `data-mage-init="{}"`attribute in phtml files.

**Comment by ihor-sviziev** [comment]
I'm having the same issue.
@sivaschenko could you set high priority for this issue?

**Comment by sivaschenko** [comment]
@DmitryFurs agree, it looks like`\Magento2\Sniffs\Legacy\PhtmlTemplateSniff` has to be split into two sniffs: one for phtml and another for PHP files. Thanks for reporting!

@ihor-sviziev done
